File: regexlib/python_re2_test_file/regexlib_239.py
# 239
# ^[a-z]+([\d_-]*[a-z]+)*@[a-z]{2,}(\.)[a-z]{2,}(\.[a-z]{2})?$
# EXPONENT
# nums:5
# EXPONENT AttackString:"a"+"a"*32+"!1 __NQ"

import re2 as re
from time import perf_counter

regex = """^[a-z]+([\d_-]*[a-z]+)*@[a-z]{2,}(\.)[a-z]{2,}(\.[a-z]{2})?$"""
REGEX = re.compile(regex)
for i in range(0, 150000):
    ATTACK = "a" + "a" * i * 1 + "!1 __NQ"
    LEN = len(ATTACK)
    BEGIN = perf_counter()
    m = REGEX.search(ATTACK)
    # m = REGEX.match(ATTACK)
    DURATION = perf_counter() - BEGIN
    print(f"{i *1}: took {DURATION} seconds!")

File: regexlib/javascript_test_file/regexlib_239.js
/* 239
 * ^[a-z]+([\d_-]*[a-z]+)*@[a-z]{2,}(\.)[a-z]{2,}(\.[a-z]{2})?$
 * EXPONENT
 * nums:3
 * EXPONENT AttackString:"a"+"a"*32+"!1 __NQ"
 */
var REGEX = /^[a-z]+([\d_-]*[a-z]+)*@[a-z]{2,}(\.)[a-z]{2,}(\.[a-z]{2})?$/
for(var i = 1; i <= 5000000; i++) {
    var time = Date.now();
    var attack_str = 'a' + 'a'.repeat(i*1) + '!1 __NQ'
    REGEX.exec(attack_str)
    // REGEX.test(attack_str);
    // attack_str.search(REGEX)
    // attack_str.match(REGEX)
    var time_cost = Date.now() - time;
    console.log(i * 1 + ": " + time_cost+" ms")
}